lib/IRGen/GenMeta.cpp

```diff
@@ -3508,9 +3490,9 @@ namespace {
       if (doesClassMetadataRequireDynamicInitialization(IGF.IGM, Target)) {
         auto classTy = Target->getDeclaredTypeInContext()->getCanonicalType();
         auto loweredClassTy = IGF.IGM.getLoweredType(classTy);
-        metadata = emitInitializeFieldOffsetVector(IGF, loweredClassTy,
-                                                   metadata,
-                                                   /*vwtable=*/nullptr);
+        emitInitializeFieldOffsetVector(IGF, loweredClassTy,
+                                        metadata,
+                                        /*vwtable=*/nullptr);
```

Comment:
So we're assuming that the superclass has been relocated already.  That's a nice simplification.  The runtime will do this?

Reply:
The relocation is handled in the next PR ("part 4").

Copying field offsets and generic arguments from higher up in the hierarchy is already implemented by `swift_initClassMetadata_UniversalStrategy()`, since we already have to do it for generic classes, where the generic arguments and field offsets are always just null in the template.